src/runtime/virtcontainers/clh.go

```diff
@@ -321,6 +321,15 @@ func (clh *cloudHypervisor) createSandbox(ctx context.Context, id string, networ
 		cache:      clh.config.VirtioFSCache,
 	}
 
+	if clh.config.EPCSize > 0 {
+		epcSection := chclient.SgxEpcConfig{
+			Size:     clh.config.EPCSize,
+			Prefault: true,
```

Comment:
Question: What does `prefault` flag do? What is the  consideration behind enabling `Prefault`  in Kata while the default value is `false` in cloud-hypervisor? /cc @sboeuf

Reply:
this is a good question, might be we need an annotation to enable or disable this
cc @mythi

Reply:
Right, I asked about this earlier too. My understanding is the setting does not matter much in Kata's case as long as the SGX EPC size comes from the k8s resources pool the device-plugin maintains.

SGX KVM doc says:
> Note that similar to regular memory, physical EPC is not allocated by mmap(), i.e. it is the responsibility of the userspace VMM to pre-allocate EPC in order to avoid out-of-EPC scenarios after a guest is running.

So the book keeping ensures each Kata sandbox gets its EPC share at the time the pod is scheduled.

The host kernel does not provide statistics about how much physical EPC is consumed so if Kata SGX pods are launched outside the k8s SGX resource model, we the bookkeeping gets out-of-sync.